woocommerce/woocommerce-ios
# Shipping Labels: order summary and woo discount section

Part of #4088 

## Description
As part of Shipping Labels M3, in this PR I implemented the order summary that displays the Subtotal, the WooCommerce discount (if available), the Order Total, a switch for completing the order, and the button for purchasing the shipping label.
I've implemented a new cell `ShippingLabelSummaryTableViewCell` that displays all the information needed when the user fulfilled all the fields in the Shipping Labels form.

## Testing
1. Make sure you have configured the WooCommerce WCShip plugin, and you have one or more payment methods set up in WP Admin under WooCommerce > Settings > Shipping > WooCommerce Shipping.
2. Launch the app in debug mode (because of the feature flag).
3. Open an order detail that is eligible for the shipping label.
4. Tap the button "Create Shipping Label".
5. Complete all the fields, and try to select the more expensive option under "Shipping Carriers and Rates" (in theory, this makes your shipping label eligible for our WooCommerce discount).
6. You should be able to see the "Shipping Label Order Summary" section at the end of the Shipping Label Form, and the "WooCommerce Discount" row should be visible.
7. Try to change the "Shipping Carriers and Rates" option, selecting the cheapest one. -> You shouldn't be able to see anymore the discount row.

PS: There is no guarantee that the cheapest carrier will not guarantee a discount, and the reverse is also true, an expensive carrier does not guarantee a discount. So do several tests if you are not able to see the discount row.


## Screenshots
| Order Summary with discount             |  Order Summary without discount |
:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 12 Pro - 2021-06-15 at 16 53 45](https://user-images.githubusercontent.com/495617/122075753-99c0ce80-cdfa-11eb-8569-ed358e464871.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-06-15 at 16 55 48](https://user-images.githubusercontent.com/495617/122075774-9d545580-cdfa-11eb-9439-203bd05b3f6a.png)




Update release notes:
